TROUBLESHOOTING.md

```diff
@@ -0,0 +1,420 @@
+
+* module.project-factory.google_service_account.default_service_account: 1 error(s) occurred:
+
+* google_service_account.default_service_account: Error creating service account: googleapi: Error 403: Identity and Access Management (IAM) API has not been used in project [seed-project-number] before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/iam.googleapis.com/overview?project=[seed-project-number] then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry., accessNotConfigured
+```
+
+**Cause:**
+
+The seed project does not have the `iam.googleapis.com` API enabled. This prevents the Project Factory from deleting the default GCE service account. 
+
+**Solution:**
+
+```
+# Requires `roles/serviceusage.serviceUsageAdmin` on $SEED_PROJECT
+gcloud services enable iam.googleapis.com --project "$SEED_PROJECT"
+```
+
+#### Missing API: `appengine.googleapis.com`
+
+**Error message:**
+
+```
+$ gcloud app describe --project [target-project-id] --format=json
+
+API [appengine.googleapis.com] not enabled on project [[seed-project-number]].
+Would you like to enable and retry (this will take a few minutes)?
+(y/N)?
+ERROR: (gcloud.app.describe) User [[service-account]] does not have permission to access app [[target-project-id]] (or it may not exist): App Engine Admin API has not been used in project [seed-project-number] before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/appengine.googleapis.com/overview?project=[seed-project-number] then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
+- '@type': type.googleapis.com/google.rpc.Help
+  links:
+  - description: Google developers console API activation
+    url: https://console.developers.google.com/apis/api/appengine.googleapis.com/overview?project=[seed-project-number]
+```
+
+**Cause**:
+
+The App Engine API is not enabled on the seed project, which prevents creation of an App Engine instance on the target project.
+
+**Solution:**
+
+```
+# Requires `roles/serviceusage.serviceUsageAdmin` on $SEED_PROJECT
+gcloud services enable appengine.googleapis.com --project "$SEED_PROJECT"
+```
+
+**Notes:**
+
+The absence of the `appengine.googleapis.com` API will not cause Terraform to fail, but any interactions with the App Engine instance will fail. Once this API is activated, it might take a few minutes for Terraform to detect that the App Engine instance is not present and then create it.
+
+- - -
+### Service account missing roles
```

Comment:
We should include a high-level link back to the [README](https://github.com/terraform-google-modules/terraform-google-project-factory/tree/adrienthebo/troubleshooting-guide#permissions).